File: androidApp/core/compose/src/main/kotlin/com/velkonost/getbetter/core/compose/components/note/completiondate/CompletionDateButton.kt
package com.velkonost.getbetter.core.compose.components.note.completiondate

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import com.velkonost.getbetter.core.compose.components.AppButton
import com.velkonost.getbetter.core.compose.components.WeightedSpacer
import com.velkonost.getbetter.core.compose.theme.Dimen.DP_16
import com.velkonost.getbetter.shared.resources.SharedR
import dev.icerock.moko.resources.compose.stringResource

@Composable
internal fun CompletionDateButton(
    label: String?,
    isLoading: Boolean,
    onCompleteClick: (() -> Unit),
    modifier: Modifier = Modifier
) {

    val viewPadding = remember { DP_16 }

    AnimatedVisibility(visible = label == null) {
        Row(
            modifier = modifier
                .padding(horizontal = viewPadding)
                .padding(bottom = viewPadding),
            verticalAlignment = Alignment.CenterVertically
        ) {
            WeightedSpacer()

            AppButton(
                labelText = stringResource(resource = SharedR.strings.note_detail_complete_goal_text),
                isLoading = isLoading,
                onClick = onCompleteClick
            )

            WeightedSpacer()
        }
    }
}

File: iosApp/iosApp/Core/Components/note/completiondate/CompletionDateButton.swift
//
//  CompletionDateButton.swift
//  iosApp
//
//  Created by velkonost on 04.11.2023.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation
import SwiftUI
import SharedSDK

struct CompletionDateButton: View {
    
    let label: String?
    let isLoading: Bool
    let onCompleteClick: () -> Void
    
    init(label: String?, isLoading: Bool, onCompleteClick: @escaping () -> Void) {
        self.label = label
        self.isLoading = isLoading
        self.onCompleteClick = onCompleteClick
    }
    
    var body: some View {
        HStack {
            if label == nil {
                Spacer()
                AppButton(
                    labelText: SharedR.strings().note_detail_complete_goal_text.desc().localized(),
                    isLoading: isLoading,
                    onClick: onCompleteClick
                )
                Spacer()
            }
        }
        .padding(.horizontal, 16)
        .padding(.bottom, 16)
        .animation(.easeInOut, value: label)
    }
}
